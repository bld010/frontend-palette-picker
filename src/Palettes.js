import React, { Component } from 'react';
import './Palettes.scss';
import MiniPalette from './MiniPalette';
import { FaTrash } from "react-icons/fa";
import PropTypes from 'prop-types'

class Palettes extends Component {
  constructor(props) {
    super(props)
    this.state = {
      reload: false
    }
  }

  handleSetPaletteEnter = (e, palette) => {
    if (e.keyCode === 13) {
      this.props.setCurrentPalette(palette)
    }
  }

  handleDeleteEnter = (e, palette) => {
    if (e.keyCode === 13){
      this.props.deletePalette(palette)
    }
  }

  // componentDidUpdate = (prevProps) => {
  //   if (this.props.folder !== prevProps.folder) {
  //     let reload = !this.setState.reload
  //     this.setState( { reload })
  //   }
  // }

  getPalettesList = () => {
   return this.props.folder.palettes.map(palette => {
      return <li key={palette.id}>
        <div class="palette-div">
          <p tabIndex={0} onKeyDown={(e) => this.handleSetPaletteEnter(e, palette)} onClick={() => this.props.setCurrentPalette(palette)}>
            {palette.name}
          </p>
          <div className="trash"> 
            <FaTrash tabIndex={0} onKeyDown={(e) => this.handleDeleteEnter(e, palette)} onClick={() => this.props.deletePalette(palette)}/>
          </div>
        </div>
        <MiniPalette tabIndex={0} onKeyDown={(e) => this.handleEnter(e, palette)} setCurrentPalette={this.props.setCurrentPalette} palette={palette}/>
      </li>
    })
  }

  render = () => {
    if (this.props.folder) {
      let { palettes } = this.props.folder

      return(
        <div className="Palettes">
        <ul>
          {!palettes.length && <li>This folder has no palettes.</li>}
          {palettes.length > 0 && this.getPalettesList()}
        </ul>
      </div>
      )
    } else {

      return(
        <div className="Palettes">
          <ul>
            <li>Select a folder to see the palettes.</li>
          </ul>
        </div>
      )
    }

  }
}

export default Palettes;


Palettes.propTypes = {
  setCurrentPalette: PropTypes.func,
  folders: PropTypes.object,
  deletePalette: PropTypes.func,
}
