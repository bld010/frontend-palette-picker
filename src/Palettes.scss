.Palettes {

  & ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;
    overflow: scroll;
    width: 100%;
    height: 28vh;
    padding: 20px 50px 20px 50px;
    min-height: 200px;
    border: 1px dashed lightgrey;
 
    

  
    & li {
      list-style-type: none;
      cursor: pointer;
      margin: 15px;
      font-family: "Open Sans", sans-serif;
      
      & .palette-div {
        width: 100%;
        display: inline-flex;
        justify-content: space-between;
        align-items: center;

        & .trash {
          opacity: 0.3;
        }
      }
    }
  }
  
}
