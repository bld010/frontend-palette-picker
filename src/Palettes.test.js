import React from "react";
import { shallow } from "enzyme";
import Palettes from "./Palettes";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";


configure({ adapter: new Adapter() });

describe("Palettes", () => {

  let mockSetCurrentPalette = jest.fn();
  let mockFolder = {
    id: 86,
    name: "Mocky",
    palettes: [
      {
        colors: [
          { hex: "#ad0c34", locked: false },
          { hex: "#74dbd4", locked: false },
          { hex: "#db9b6b", locked: false },
          { hex: "#9863f9", locked: false },
          { hex: "#f9dc98", locked: false }
        ]
      }
    ]
  };
  let mockDeleteFolder = jest.fn();
  let wrapper = shallow(
    <Palettes
      setCurrentPalette={mockSetCurrentPalette}
      folder={mockFolder}
      deleteFolder={mockDeleteFolder}
    />
  );


  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSetPaletteEnter on the keydown', () => {
    wrapper.instance().handleSetPaletteEnter = jest.fn()
    wrapper.find('p').simulate('keydown', {keyCode: 13})

    expect(wrapper.instance().handleSetPaletteEnter).toHaveBeenCalled()
  })

  it('should call handleDeleteEnter on the keydown', () => {
    wrapper.instance().handleDeleteEnter = jest.fn()
    wrapper.find('.trash').simulate('keydown', {keyCode: 13})

    expect(wrapper.instance().handleSetPaletteEnter).toHaveBeenCalled()
  })
});
