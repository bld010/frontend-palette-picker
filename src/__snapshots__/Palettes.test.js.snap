// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes should match the snapshot 1`] = `ShallowWrapper {}`;
